parachain/primitives/core/src/inbound.rs

```diff
@@ -5,23 +5,23 @@
 use codec::{Decode, Encode};
 use frame_support::PalletError;
 use scale_info::TypeInfo;
-use sp_core::H256;
+use sp_core::{H160, H256};
 use sp_runtime::RuntimeDebug;
 use sp_std::vec::Vec;
 
 /// A trait for verifying inbound messages from Ethereum.
 pub trait Verifier {
-	fn verify(message: &Message) -> Result<(), VerificationError>;
+	fn verify(event: &Log, proof: &Proof) -> Result<(), VerificationError>;
```

Comment:
Maybe not direct relevant to change in this PR, I would assume event log is not required for the verification, with the proof and `header.receipts_root` we can generate the `receipt` which internally contains the log event.

Reply:
Also we have duplicated definition for `Log` structure in 

https://github.com/Snowfork/snowbridge/blob/6b592235b907e00e9fe6ab4446ff48564bd41474/parachain/pallets/inbound-queue/src/envelope.rs#L12C1-L17

and 

https://github.com/Snowfork/snowbridge/blob/6b592235b907e00e9fe6ab4446ff48564bd41474/parachain/primitives/ethereum/src/log.rs#L8-L13

Maybe we can make some furthur refactoring.

Reply:
Yeah, that's a good observation. I saw that as well while doing this PR. 

However I think its easier to troubleshoot problems if the `Log` is submitted as well. Otherwise if there are issues, we would have to decode the `Proof` manually to extract the Log. Which brings us full circle to the problem this PR is solving - improving debuggability.